packages/blade/src/hooks/useExposeInputRef.ts

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import type { TextInput as TextInputReactNative } from 'react-native';
+
+type NativeHTMLInputRefProps = Omit<HTMLInputElement, keyof HTMLElement> & {
+  focus: HTMLElement['focus'];
+};
+
+/**
+ * A hook which only exposes the properties of html input element via imparative hook
+ *
+ * It avoids exposing other native properties of HTMLElement
+ * like `style` `classList` to avoid unintended usage of refs.
+ */
+const useExposeInputRef = (
+  targetRef: React.ForwardedRef<NativeHTMLInputRefProps>,
+  inputRef: React.RefObject<TextInputReactNative | HTMLInputElement>,
+) => {
+  React.useImperativeHandle(
+    targetRef,
+    (): NativeHTMLInputRefProps => {
+      const input = inputRef.current as HTMLInputElement & {
+        selectedOptions: HTMLSelectElement['selectedOptions'];
+      };
+      return {
```

Comment:
let's limit this to only the ones we know we need it? as an when requirement comes in we can add other options?

Reply:
Yeah technically for the usecases of the PG, we need to expose focus only. 

I checked the codebase, and it seems that if you want to use uncontrolled input via the `register()` method of hook form, then you need to expose the other input related props. We can remove other props too.